stdlib/public/core/StringBreadcrumbs.swift

```diff
@@ -0,0 +1,139 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+
+// @opaque
+internal final class _StringBreadcrumbs {
+  static var breadcrumbStride: Int { return 32 }
+
+  var utf16Length: Int
+
+  // TODO: does this need to be a pair?.... Can we be smaller than Int?
+  var crumbs: [String.Index]
```

Comment:
Why is `crumbs` an Array instead of tail allocated, fixed sized, trivial storage. Don't we want the _StrigBreadcrumbs implicit `deinit` to be a nop? (I realize this is non-ABI impacting, hence not important. Just bringing it up while I noticed).

Reply:
Sure, that would work as an optimization. This whole class is an optional second tail allocation on String itself, but this would speed up access and destruction.